bson.cpp

```diff
@@ -1064,6 +1145,49 @@ void parseTypeMap(hippo_bson_conversion_options_t *options, const Array &typemap
 			options->array_class_name = array_type;
 		}
 	}
+
+	if (typemap.exists(s_types)) {
+		if (!typemap[s_types].isArray()) {
+			throw MongoDriver::Utils::throwInvalidArgumentException("The 'types' key in the type map should be an array");
+		}
+
+		for (ArrayIter iter(typemap[s_types]); iter; ++iter) {
+			Variant key(iter.first());
+			String s_key = key.toString();
+
+			const Variant& data(iter.secondRef());
+			if (!data.isString()) {
+				throw MongoDriver::Utils::throwInvalidArgumentException("The typemap for type '" + s_key + "' should be a string");
+			}
+			String s_type_class = data.toString();
+
+			/* Depending on the key, set the right option (or throw an exception) */
+			if (CASECMP(s_key, s_MongoBsonBinary_shortName)) {
+				validateTypeWrapperClass(s_type_class);
+				options->types.binary_class_name = s_type_class;
```

Comment:
I am not, and I also don't see why I should. The TypeWrapped class could even be returning a scalar as we discussed earlier.